src/Infrastructure/Transport/FileWatcherFactory.php

```diff
@@ -26,7 +26,7 @@ public function createTransport(string $dsn, array $options = [], ?SerializerInt
                 $dsn->directory,
                 $this->gitDirectoriesLocation
                     ->sub('file-watcher-git')
-                    ->sub(\sha1($dsn->directory->path) . '.git'),
+                    ->sub(\sha1((string) $dsn->directory->path) . '.git'),
```

Comment:
`dsn`, `directory` and `path` are never null?

Reply:
After checking the [code snippet](https://github.com/michaelpetri/php-git/blob/main/src/Value/Directory.php#L12), I think it   always has the string type.